### PR TITLE
For #27540: Added screen size configuration change handling 

### LINKED ISSUE #27540: [Bug]: search bar becomes very small after fold from large to small screen
### Steps to reproduce

1. Obtain foldable device (occurs 100% of the time on fold 4)
2. Open Firefox in unfolded mode and open a website (only happens if app is started in unfolded mode)
3. Close phone
4. Search bar is now the wrong size

### Expected behaviour

The search bar should resize to the proper size as if app is opened in folded mode 

### Actual behaviour

The searchbar is very small, with the lock icon detailing website security taking up most of the space. Upon clicking on visible part of searchbar (not very easy), original size and function is restored and now works properly when folded

### Device name

Samsung Galaxy Z Fold 4

### Android version

Android 12L/13

### Firefox release type

Firefox

### Firefox version

106.1.0

### Device logs

_No response_

### Additional information

_No response_

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21867)

### PR DESCRIPTION
**Pull request by zmckenney**
On foldable devices or tablets when using multiple applications, fenix will not adjust the toolbar for a new minimum width after initial launch. Handling configuration changes after initializing UI is required to handle the adding and removing of tablet actions on the toolbar.  Incorrect states shown below:

Going from a large size to a small size:
large-to-small-size

Going from a small to large size:
small-to-large-size

Opening a side-by-side app (same scenario as large to small):
sidebyside

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27540

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -71,7 +77,6 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
                 ),
             )
         }
-
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:74]: nit: let's not remove this new line

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 2]
```diff
@@ -243,6 +169,129 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
...
+        this.isTablet = isTablet
+    }
+
+    @Suppress("LongMethod")
+    private fun addTabletActions(context: Context) {
+        val enableTint = ThemeManager.resolveAttribute(R.attr.textPrimary, context)
+        val disableTint = ThemeManager.resolveAttribute(R.attr.textDisabled, context)
+        backAction?.let {
+            browserToolbarView.view.addNavigationAction(it)
+        } ?: kotlin.run {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:190]:
```suggestion
        } ?: run {
```
We can shorten this to `run` for all your actions here. I am mostly asking for this change because we use `run` everywhere else.

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 3]
```diff
@@ -243,6 +169,129 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
...
+        }
+
+        this.isTablet = isTablet
+    }
+
+    @Suppress("LongMethod")
+    private fun addTabletActions(context: Context) {
+        val enableTint = ThemeManager.resolveAttribute(R.attr.textPrimary, context)
+        val disableTint = ThemeManager.resolveAttribute(R.attr.textDisabled, context)
+        backAction?.let {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:188]: Let's add a new line above this. Since the `backAction` is rather big, it could use the separation for easier readability.

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 4]
```diff
@@ -243,6 +169,129 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
...
+                        ToolbarMenu.Item.Forward(viewHistory = true),
+                    )
+                },
+                listener = {
+                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Forward(viewHistory = false),
+                    )
+                },
+            ).also {
+                browserToolbarView.view.addNavigationAction(it)
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:240]: **[Open Question]** It's a bit odd to see that we have to repeat `browserToolbarView.view.addNavigationAction(it)` twice in these blocks.
I am wondering if it would just be simpler to just do the following:

```
if (forwardAction == null) {
  forwardAction = BrowserToolbar.TwoStateButton(...)
}

forwardAction?.let {
    browserToolbarView.view.addNavigationAction(it)
}
```

**Review by gabrielluong — APPROVED** [review]
@zmckenney I do appreciate seeing multiple commits. It's definitely easier to review and approve individual commits in mozilla-central using phabricator as compare to GitHub reviews. I would like to avoid landing commits where we are doing code cleanup in our GitHub repo. If it's okay with you, I would prefer if we did a squash landing or you could squash your changes while addressing the feedback. 

When working with multiple commits especially in phabricator, the expectation is that the individual commits are amended to address feedback. That way each commit is serving as its own unit of work (eg, Refactor, Implementing X, etc). 

Lastly, I am wondering if Product is tracking this anywhere. It would be good to inform them that this is being fixed.

Thanks

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt (hunk 4 above)
Comment (zmckenney) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:240]: ya, I agree it's better to just leave it a simple null check and set

**Comment by zmckenney** [comment]
Squashed into single commit while updating from feedback.
